### PR TITLE
feat: show Block Editor Side on Traditional

### LINKED ISSUE #28198: UVE: Bring back inline editing for block editor fields
### Parent
#27546 

### Task description
As a web content editor using dotCMS, I want to have the inline editing feature available in the new editor, so that I can easily make quick edits to content without having to switch to the code view.

Wait for #28196

https://github.com/dotCMS/core/assets/751424/1031f189-37ee-4aba-b2b1-eb8030fd46f0

### Acceptance Criteria
1. The inline editing feature is successfully integrated into the new editor.
2. Users can seamlessly edit content inline without any issues.
3. The functionality works the same as it did in the old editor.
4. Users are not forced to update their code when using the inline editing feature.

### External Links
- Slack conversation: #dotCMS-development
- Support ticket: #12345

### Assumptions & Initiation Needs
- The new editor interface has the capability to support the inline editing feature.
- Users are familiar with how the inline editing feature worked in the old editor.
- A thorough testing phase will be conducted to ensure the feature works as expected.

### Quality Assurance Notes & Workarounds
- QA team should test the feature extensively to ensure it works seamlessly.
- Any issues or bugs encountered during testing should be documented and addressed before release.

### Sub-Tasks & Estimates
1. Research and evaluate the feasibility of bringing back inline editing feature

### PR DESCRIPTION
**Pull request by rjvelazco**
This pull request includes multiple changes to enhance the block editor functionality and improve the workflow actions in the `core-web` project. The most important changes include the addition of a new block editor sidebar component, updates to the workflow actions service, and modifications to the event handling and test cases.

### Enhancements to Block Editor Functionality:

* Added a new `DotBlockEditorSidebarComponent` with its HTML, SCSS, and TypeScript files to handle the block editor sidebar functionality. This component includes methods for saving editor changes, closing the sidebar, and handling events. [[1]](diffhunk://#diff-5490d402dcc3eb8dca39a3c27a2a45b3338bea2a103206240dbc0134f6a9b4ecR1-R35) [[2]](diffhunk://#diff-bd12ae9c296a163014328299a9345bd409b8f203772fd8dc954502841871f874R1-R43) [[3]](diffhunk://#diff-1513562a5281a737d81d7b060a1f2663f57aed0934c633f947cddf3af61d56bbR1-R170)
* Updated `edit-ema-editor.component.html` to include the new block editor sidebar component and handle the `onSaved` event to reload the store.

### Improvements to Workflow Actions Service:

* Modified `dot-workflow-actions-fire.service.ts` to use URLSearchParams for appending query parameters and refactored the handling of `inode` and `indexPolicy`.
* Updated test cases in `dot-workflow-actions-fire.service.spec.ts` to reflect changes in the workflow actions fire service and ensure proper handling of the `indexPolicy` parameter. [[1]](diffhunk://#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL185-R185) [[2]](diffhunk://#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL198-R197) [[3]](diffhunk://#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL218-R217) [[4]](diffhunk://#diff-edfb8c91327f7780ddeeaa0d3fd01655756ebdc77197088c6973460e0c38ddfaL236-R234)

### Event Handling and Test Cases:

* Updated `dot-events.service.ts` to cast the filtered observable as `Observable<DotEvent<T>>`.
* Added new test cases for the `DotBlockEditorSidebarComponent` to ensure correct behavior of the sidebar, block editor inputs, save functionality, and error handling.
* Updated `edit-ema-editor.component.spec.ts` to include the new block editor sidebar component and handle the inline edit block editor event. [[1]](diffhunk://#diff-34ddc5fbacaf04b962f2037385ed284310d5faf35ba409d5705b2caadd5d796aR31) [[2]](diffhunk://#diff-34ddc5fbacaf04b962f2037385ed284310d5faf35ba409d5705b2caadd5d796aR82-R85) [[3]](diffhunk://#diff-34ddc5fbacaf04b962f2037385ed284310d5faf35ba409d5705b2caadd5d796aL111-R119)

### Minor Changes:

* Added a method to inject the inline block editor in `dot-edit-content-html.service.ts`.

### Video

https://github.com/user-attachments/assets/e6d59b8a-119a-4f93-87ae-44148541b86d

### Handle Error

https://github.com/user-attachments/assets/8c606cce-f562-4fca-a9eb-50c6082eb383

This PR fixes: #28198

### CODE REVIEW

File: core-web/libs/data-access/src/lib/dot-events/dot-events.service.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class DotEventsService {
         // TODO: need to make this method to support multiple events
         return this.subject
             .asObservable()
-            .pipe(filter((res: DotEvent<T>) => res.name === eventName));
+            .pipe(filter((res) => res.name === eventName)) as Observable<DotEvent<T>>;
```
Comment (fmontes) [inline core-web/libs/data-access/src/lib/dot-events/dot-events.service.ts:27]: why `as`? are we typing this correctly?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-block-editor-editing/dot-block-editor-sidebar.component.ts
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-block-editor-editing/dot-block-editor-sidebar.component.ts:1]: Didn't we already have this component?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts [hunk 2]
```diff
@@ -67,23 +67,51 @@ export class InlineEditService {
...
+                        {
+                            action: 'editor-inline-editing',
+                            payload: { dataset }
+                        },
+                        '*'
+                    );
+                });
+            });
+        });
+        `;
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts:110]: Can we just use the block editor Angular component directly?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/data-access/src/lib/dot-events/dot-events.service.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/data-access/src/lib/dot-events/dot-events.service.ts:27]: Typescript error. I didn't dig into it, not part of the PR but it was throwing a warning

File: core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts:110]: Mmm... Didn't get that question.

This is for traditional pages. We need to track when someone clicks on the Block Editor content and notify the UVE of that action. This approach is similar to how we handle `WYSIWYG` content.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-block-editor-editing/dot-block-editor-sidebar.component.ts
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-block-editor-editing/dot-block-editor-sidebar.component.ts:1]: I copied the old component, moved it to a new folder, and improved the code to reduce its dependency on the old editor, as the other component was tied to it

File: core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts:110]: I moved this to a separate file; however, all scripts required for traditional pages should be inside the `sdk-editor-vtl.js` file.

Comment (fmontes) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts]: Traditional or not the block editor loads in the angular side... so it should be a component.

Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts]: I think this whole code, should be on our SDK and make it as reusable as possible for both traditional and headless

Comment (fmontes) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts]: I already have a customer complaning that the tinmymce needs to be in their building even when they're not using it.

The block editor doesn't load in the customer page it loads in a primeng sidepanel in UVE.

Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts]: We already discussed this!

Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/services/inline-edit/inline-edit.service.ts]: **FYI**: 

1. This was not the block editor but a script to notify the UVE that a content was clicked
2. It's going to be moved to the SDK.vtl - I already have added comments about that.

**Review by fmontes — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts [hunk 3]
```diff
@@ -998,6 +1008,20 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
...
+                        header: this.dotMessageService.get(
+                            'dot.common.license.enterprise.only.error'
+                        ),
+                        message: this.dotMessageService.get('editpage.not.lincese.error')
+                    });
+
+                    return;
+                }
+
+                this.#dotEventsService.notify(INLINE_EDIT_BLOCK_EDITOR_EVENT, payload);
```
Comment (zJaaal) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:1024]: Why this is not managed from the store? We can create a `computed` and a method to listen and set the current state of this sidebar 

I think is better to have that info in the store or in an feature of the store, that way we can control it if needed and we don't add another moving piece to the equation, this service

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts (hunk 3 above)
Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts:1024]: Maybe not a new store feature, but update the editor one, to handle this property

**Review by zJaaal — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**